tests/test_oml/test_functional/test_metrics/test_topological_metrics.py

```diff
@@ -0,0 +1,124 @@
+from typing import Callable, Dict, Tuple
+
+import pytest
+import torch
+
+from oml.functional.metrics import PCA, TMetricsDict, calc_pcf, calc_topological_metrics
+from oml.utils.misc import remove_unused_kargs
+
+
+def compare_dicts_recursively(d1: Dict, d2: Dict) -> bool:  # type: ignore
```

Comment:
By the way, why don't you use `d1 == d2`? Python is able to compare dicts recursively. The popular reason to do it manually is to check that values are close (`torch.close`), but not the same. Seems like it's not your case

Reply:
I know, but I want to know exactly which keys differs. There is very similar code [here](https://github.com/OML-Team/open-metric-learning/blob/aaf6142c8bf737b576cc0c8cf43099bf9f62e39b/tests/test_oml/test_metrics/test_embedding_metrics.py#L24) but as for me it doesn't help much when debugging is required.

I don't like the duplicated code, but I don't want to touch that test module either. If you think it is ok, I can replace the code in `test_embedding_metrics.py` by this one. Where is it better to put this common function then?

Reply:
Okay, I see your point and agree. 
Then, put the function to utils and use it in both cases: your current tests and `test_embedding_metrics.py`